File: sitzungsbeleg/tests/test_umgebung.mjs
// Reine-Funktions-Test fuer die Achse Umgebung (C12, Auftrag 2026-08-28 12:45): Projekt-
// Zellen-Chip (kein Chip bei entwicklung, Kuerzel bei Abweichung) + Sichtbarkeitsregel des
// Seitenleisten-Blocks ("erscheint nur, wenn im Zeitraum >= 2 Werte vorkommen") -- Muster
// test_chat_sicht.mjs: reine Funktionen rein, kein DOM/Browser noetig.
// Ausfuehren: `node scripts/sitzungsbeleg/tests/test_umgebung.mjs`.
import assert from "node:assert/strict";
import { test } from "node:test";

import { umgebungChipHtml, umgebungBlockSichtbar } from "../static/js/format.js";

test("umgebungChipHtml: kein Chip bei entwicklung", () => {
  assert.equal(umgebungChipHtml("entwicklung"), "");
});

test("umgebungChipHtml: kein Chip bei fehlendem/unbekanntem Wert", () => {
  assert.equal(umgebungChipHtml(""), "");
  assert.equal(umgebungChipHtml(undefined), "");
  assert.equal(umgebungChipHtml("staging"), "");
});

test("umgebungChipHtml: Kuerzel Abn. bei abnahme", () => {
  var html = umgebungChipHtml("abnahme");
  assert.match(html, /umgebung-chip umgebung-abnahme/);
  assert.match(html, />Abn\.</);
});

test("umgebungChipHtml: Kuerzel Betrieb bei betrieb", () => {
  var html = umgebungChipHtml("betrieb");
  assert.match(html, /umgebung-chip umgebung-betrieb/);
  assert.match(html, />Betrieb</);
});

test("umgebungBlockSichtbar: ein Wert im Zeitraum -> Block bleibt versteckt", () => {
  assert.equal(umgebungBlockSichtbar([{ name: "entwicklung", anzahl: 12 }]), false);
});

test("umgebungBlockSichtbar: zwei Werte im Zeitraum -> Block sichtbar", () => {
  assert.equal(
    umgebungBlockSichtbar([{ name: "entwicklung", anzahl: 12 }, { name: "abnahme", anzahl: 3 }]),
    true,
  );
});

test("umgebungBlockSichtbar: ein Wert mit anzahl=0 zaehlt nicht mit", () => {
  assert.equal(
    umgebungBlockSichtbar([{ name: "entwicklung", anzahl: 12 }, { name: "betrieb", anzahl: 0 }]),
    false,
  );
});

test("umgebungBlockSichtbar: leere/fehlende Liste -> versteckt", () => {
  assert.equal(umgebungBlockSichtbar([]), false);
  assert.equal(umgebungBlockSichtbar(undefined), false);
});

File: sitzungsbeleg/tests/test_umgebung.py
"""Achse Umgebung (C12): reine Funktionen `wert`/`kuerzel`/`verteilung`. Kein SQL/DB hier --
das deckt tests/test_web.py (API-Filter/Distribution) und tests/test_main.py (Stop-Hook-Ableitung)."""
from __future__ import annotations

import unittest

from .. import umgebung


class WertTest(unittest.TestCase):
    def test_bekannter_wert_bleibt_unveraendert(self) -> None:
        self.assertEqual(umgebung.wert("abnahme"), "abnahme")
        self.assertEqual(umgebung.wert("betrieb"), "betrieb")
        self.assertEqual(umgebung.wert("entwicklung"), "entwicklung")

    def test_fehlender_wert_ist_entwicklung(self) -> None:
        self.assertEqual(umgebung.wert(None), "entwicklung")
        self.assertEqual(umgebung.wert(""), "entwicklung")

    def test_unbekannter_wert_faellt_auf_entwicklung_zurueck(self) -> None:
        self.assertEqual(umgebung.wert("staging"), "entwicklung")


class KuerzelTest(unittest.TestCase):
    def test_entwicklung_zeigt_keinen_chip(self) -> None:
        self.assertEqual(umgebung.kuerzel("entwicklung"), "")

    def test_abnahme_zeigt_abn(self) -> None:
        self.assertEqual(umgebung.kuerzel("abnahme"), "Abn.")

    def test_betrieb_zeigt_betrieb(self) -> None:
        self.assertEqual(umgebung.kuerzel("betrieb"), "Betrieb")

    def test_unbekannter_wert_zeigt_keinen_chip(self) -> None:
        self.assertEqual(umgebung.kuerzel("irgendwas"), "")


class VerteilungTest(unittest.TestCase):
    def test_zaehlt_je_wert_alphabetisch(self) -> None:
        self.assertEqual(
            umgebung.verteilung(["entwicklung", "abnahme", "entwicklung"]),
            [{"name": "abnahme", "anzahl": 1}, {"name": "entwicklung", "anzahl": 2}],
        )

    def test_leere_liste_bleibt_leer(self) -> None:
        self.assertEqual(umgebung.verteilung([]), [])
